Torch/TorchBase.cs

```diff
@@ -125,17 +142,13 @@ public bool IsOnGameThread()
                     await Task.Run(() =>
                     {
                         if (e.WaitOne(60000))
-                            return;
-
-                        Log.Error("Save failed!");
-                        Multiplayer.SendMessage("Save timed out!", "Error");
+                            callback(SaveGameStatus.Success);
+                        return;
```

Comment:
Double check this, `return;` is not in the conditional and the callback will never be called with `SaveGameStatus.TimedOut`.